docs/languages/csharp/expression-trees-interpreting.md

```diff
@@ -0,0 +1,579 @@
+# Interpreting Expressions
+
+Now, let's write some code to examine the structure of an 
+*expression tree*. Every node in an expression tree will be
+an object of a class that is derived from `Expression`.
+
+That design makes visiting all the nodes in an expression tree
+a relatively straight forward recursive operation. The general strategy
+is to start at the root node and determine what kind of node it is.
+
+If the node type has children, recursively visit the children. At each
+child node, repeat the process used at the root node: determine the
+type, and if the type has children, visit each of the children.
+
+## Examining an Expression with No Children
+Let's start by visiting each node in a very simple expression tree.
+Here's the code that creates a constant expression and then
+examines its properties:
+
+```cs
+var constant = Expression.Constant(24, typeof(int));
+
+Console.WriteLine($"This is an {constant.NodeType} expression type");
+Console.WriteLine($"The type of the constant value is {constant.Type}");
+Console.WriteLine($"The value of the constant value is {constant.Value}");
+```
+
+This will print the following:
+
+```
+This is an Constant expression type
+The type of the constant value is System.Int32
+The value of the constant value is 24
+```
+
+Now, let's write the code that would examine this expression and write
+out some important properties about it. Here's that code:
+
+## Examining a simple Addition Expression
+
+Let's start with the addition sample from the
+introduction to this section.
+
+```cs
+Expression<Func<int>> sum = () => 1 + 2;
+```
+
+> I'm not using `var` to declare this expression tree. You can't use
```

Comment:
Although good info, this aside is too large and affects the flow of reading the article.  I'd suggest writing a blurb about this somewhere and linking to it.  I'd write it like this:

> Note that `var` is not used to declare this expression tree, as it is not possible because the right-hand side of the expression is implicitly typed.  To understand this more deeply, [read here](example.com)